### PR TITLE
EVG-12554 handle aliases before creating tasks

### PR DESCRIPTION
**Pull request by ablack12**
Because in `createTasksForBuild` we strictly match on alias, we filter out any dependencies that don't match the alias. Some functions that call this already add dependencies to the list of taskNames, but some don't seem to so I added `IncludePatchDependencies`. 

I also snuck in a performance fix for BlockedState, because this individually finds each of the tasks dependencies, but getTaskDependencies already gets all of these in one statement so we can just use that.

### CODE REVIEW

File: model/lifecycle.go [hunk 1]
```diff
@@ -653,8 +653,13 @@ func CreateBuildFromVersionNoInsert(args BuildCreateArgs) (*build.Build, task.Ta
 	}
 	b.BuildNumber = strconv.FormatUint(buildNumber, 10)
 
+	// no tasks available for build
+	if len(args.Aliases) > 0 && len(args.TaskNames) == 0 {
```
Comment (john-m-liu) [inline model/lifecycle.go:657]: would it be better to check this earlier on in https://github.com/evergreen-ci/evergreen/blob/c8d081a3e7ab6fe37d1b952e1e626f6cc76253db/model/patch_lifecycle.go#L397? you already do this in the repotracker

File: repotracker/repotracker.go [hunk 2]
```diff
@@ -788,11 +790,36 @@ func createVersionItems(ctx context.Context, v *model.Version, metadata model.Ve
 			if !match {
 				continue
 			}
+			for _, t := range buildvariant.Tasks {
+				match, err := aliases.HasMatchingTask(buildvariant.Name, buildvariant.Tags, projectInfo.Project.FindProjectTask(t.Name))
+				if err != nil {
+					grip.Error(message.WrapError(err, message.Fields{
+						"message": "error creating tasks with alias filter",
```
Comment (john-m-liu) [inline repotracker/repotracker.go:797]: nit: finding rather than creating?

File: model/lifecycle.go [hunk 3]
```diff
@@ -737,21 +742,6 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 	}
 
 	for _, task := range buildVariant.Tasks {
-		if aliases != nil {
```
Comment (john-m-liu) [inline model/lifecycle.go:740]: This is probably a result of me not understanding the problem fully, but don't you still need to have this check in some form for patches? Or are patch tasks already guaranteed to match the provided alias once they reach here?

File: model/lifecycle.go (hunk 1 above)
Comment (ablack12) [reply model/lifecycle.go:657]: Good point. I think I'd rather put the check higher up in this function and remove it from the repotracker, just to ensure we have a catch-all (although it looks like only the repotracker populates Aliases, but that may change for whatever reason so)

File: repotracker/repotracker.go (hunk 2 above)
Comment (ablack12) [reply repotracker/repotracker.go:797]: truee

File: model/lifecycle.go (hunk 3 above)
Comment (ablack12) [reply model/lifecycle.go:740]: Patch tasks will already matched the provided alias before calling createTasksForBuild, and added dependencies, before constructing the taskNames slice, so using just taskNames is correct as it doesn't filter out dependencies.

For reference, createTasksForBuild is called from [AddTasksToBuild](https://github.com/evergreen-ci/evergreen/blob/master/model/lifecycle.go#L561), where Aliases was already unpopulated, and from [CreateBuildFromVersionNoInsert](https://github.com/evergreen-ci/evergreen/blob/master/model/lifecycle.go#L657), which did populate Aliases, but now we handle it beforehand (in the repotracker, as the other callers of this function don't have aliases).

**Pull request merged**